proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
-* Mobile app will be responsible for decoding the String value of vehicle data into a usable form by applying conversions as defined by OEM proprietary spec sheet for corresponding vehicle data.
-* Response will also contain an optional “info” field. This field contains additional human readable information regarding the request/response
-
+Sample response from module:
+```json
+{
+	"rpm":844
+}
 ```
-<function name="OnGenericNetworkData" functionID="OnGenericNetworkDataID" messagetype="notification">
-    <description>Callback for the periodic and non-periodic generic vehicle data read function.</description>
-    
-	<param name="networkData" type="NetworkData" mandatory="false" minsize="1" maxsize="500" array="true">
-      <description>
-      	Returned Network Data value
-      </description>
-    </param>  
-
-	<param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
-      <description>Provides additional human readable info regarding the notification.</description>
-    </param>	
-	
-</function>
+Structure in API:
 ```
-
-## Potential downsides
-
-* This approach has underlying assumption that Mobile app would have access to OEM specific proprietary interface details.
-  * This is acceptable since we target only OEM proprietary Mobile apps to utilize this feature. Moreover, benefits of access to more vehicle data params outweigh this limitation.
-* Mobile app is responsible for maintaining the network definitions as well as decoding the raw value received from vehicle.
-  * Alternate way would be to maintain all the network signal definitions at head unit side. Which means it might not be scalable with any changes/additions to network signal definitions with introduction of newer vehicle architectures/features. On the contrary, Mobile app will maintain only the required network signal definitions, which will be significantly smaller than entire suite of network definitions.
-
-## Impact on existing code
-
-SDL needs to track requests by NetworkDataIdentifier per app and work through the business logic of duplicate requests. SDL would also need to distribute the received data to apps which have subscribed. 
-
-## Alternatives considered
-
-* Add additional parameter in existing GetVehicleData/SunscribeVehicleData/UnsubscribeVehicleData RPCs:
-  * networkDataRequest - which specifies paramId and NetworkDataAttribute String array to identify the network data item
-
+<param name="rpm" type="Integer" minvalue="0" maxvalue="20000" mandatory="false">
+	<description>The number of revolutions per minute of the engine</description>
+</param>
+```
+* Structure defines the data type as Integer and a reference as OEM_REF_RPM
+* OEM_REF_RPM (or a more suitable nomenclature)
+  * This acts as a key for OEM side mapping to signal information associated with RPM. 
+  * The mapping table for this reference is purely at OEM side. OEM may choose to implement it at HMI layer, fetch it over the cloud or both.
```

Comment:
Change to:
The mapping table for this reference is purely **on the** OEM side. **OEMs** may choose to implement it **on the** HMI layer, fetch it over the cloud or both.

Reply:
Please update this.